src/distrs/norm.jl

```diff
@@ -82,17 +82,25 @@ normlogccdf(z::Number) = z > 1.0 ?
 function normlogccdf(μ::Real, σ::Real, x::Number)
     if iszero(σ) && x == μ
         z = zval(zero(μ), σ, one(x))
-    else        
-        z = zval(μ, σ, x)        
+    else
+        z = zval(μ, σ, x)
     end
     normlogccdf(z)
 end
 
 norminvcdf(p::Real) = -erfcinv(2*p) * sqrt2
-norminvcdf(μ::Real, σ::Real, p::Real) = xval(μ, σ, norminvcdf(p))
+function norminvcdf(μ::Real, σ::Real, p::Real)
+    # Promote to ensure that we don't compute erfcinv in low precision and then promote
+    _μ, _σ, _p = map(float, promote(μ, σ, p))
```

Comment:
Couldn't you just:
```suggestion
    _μ, _σ, _p = promote(μ, σ, float(p))
```
?

Reply:
No, this will cause undesired promotions: If e.g. `mu` and `sigma` are of type `Float32` and `p` is of type `Int` or `Rational` (more likely, eg. in Distributions fallback for `median` uses `1//2`), then the PR would promote to and return `Float32` whereas your suggestion would promote to and return `Float64`.

Reply:
Ah, good point.

Reply:
The `float` could be removed though, I just copied it from Distributions but SpecialFunctions calls it already: https://github.com/JuliaMath/SpecialFunctions.jl/blob/414385a7097e0f376b1273a3e0deac9d30cf0f9e/src/erf.jl#L397

Reply:
I simplified the promotion.